authentik/sources/saml/models.py

```diff
@@ -182,11 +191,35 @@ def serializer(self) -> type[Serializer]:
         return SAMLSourceSerializer
 
     @property
-    def icon_url(self) -> str:
-        icon = super().icon_url
-        if not icon:
-            return static("authentik/sources/saml.png")
-        return icon
+    def property_mapping_type(self) -> type[PropertyMapping]:
+        return SAMLSourcePropertyMapping
+
+    def get_base_user_properties(self, root: Any, name_id: Any, **kwargs):
+        attributes = {}
+        assertion = root.find(f"{{{NS_SAML_ASSERTION}}}Assertion")
+        if assertion is None:
+            raise ValueError("Assertion element not found")
+        attribute_statement = assertion.find(f"{{{NS_SAML_ASSERTION}}}AttributeStatement")
+        if attribute_statement is None:
+            raise ValueError("Attribute statement element not found")
+        # Get all attributes and their values into a dict
+        for attribute in attribute_statement.iterchildren():
+            key = attribute.attrib["Name"]
+            attributes.setdefault(key, [])
+            for value in attribute.iterchildren():
+                attributes[key].append(value.text)
+        # Flatten all lists in the dict
+        for key, value in attributes.items():
+            attributes[key] = BaseEvaluator.expr_flatten(value)
+        attributes["username"] = name_id.text
+
+        # TODO: populate this
+        attributes["groups"] = []
```

Comment:
@BeryJu could you provide guidance on how to populate this? It should be a list of group IDs, that is used below to construct group properties.